src/server/cluster/outgoing_slot_migration.h

```diff
@@ -53,10 +53,21 @@ class OutgoingMigration : private ProtocolClient {
     return migration_info_;
   }
 
-  std::string GetErrorStr() const {
+  void ReportError(dfly::GenericError err) ABSL_LOCKS_EXCLUDED(error_mu_) {
```

Comment:
I find the naming here a bit confusing, you already reported the error to the execution state , now you just save last error .  I would call it OnErrorReset or similar and will do the 
cntx_.GetError();
and the cntx_.Reset(nullptr); 
in this function